Lib/fontTools/colorLib/builder.py

```diff
@@ -0,0 +1,54 @@
+from fontTools.ttLib import newTable
+from .errors import ColorLibError
+
+
+def buildCOLR(colorLayers):
+    """Build COLR table from color layers mapping.
+
+    Args:
+        colorLayers: Dict[str, List[Tuple[str, int]]]: map of base glyph names (str)
+            to lists of layer glyph names (str) and palette indices (int) tuples.
+
+    Return:
+        A new COLRv0 table.
+    """
+    from fontTools.ttLib.tables.C_O_L_R_ import LayerRecord
+
+    colorLayerLists = {}
+    for baseGlyphName, layers in colorLayers.items():
+        colorLayerLists[baseGlyphName] = [
+            LayerRecord(layerGlyphName, colorID) for layerGlyphName, colorID in layers
+        ]
+
+    colr = newTable("COLR")
+    colr.version = 0
+    colr.ColorLayers = colorLayerLists
+    return colr
+
+
+def buildCPAL(palettes):
+    """Build CPAL table from list of color palettes.
+
+    Args:
+        palettes: List[List[Tuple[float, float, float, float]]]: list of lists
```

Comment:
Maybe use typing hints directly?

Reply:
yes, that'd be nice for self-documenting purposes, but we don't run any static typechecker. Also, otTables classes are dynamically generated and mypy doesn't seem to like that very much (#1725). Let's do fix separately some other time

Reply:
Given that docstrings aren't checked either, I'd prefer we actually use type hints so that adding mypy is less work later. My 2 cents :)

Reply:
fair enough, I added the type hints.